File: apps/java/src/main/java/com/functions/IdentityProbeFunction.java
package com.functions;

import com.microsoft.azure.functions.*;
import com.microsoft.azure.functions.annotation.*;
import java.util.Optional;

/**
 * Identity probe — checks managed identity availability.
 */
public class IdentityProbeFunction {

    @FunctionName("identityProbe")
    public HttpResponseMessage run(
            @HttpTrigger(
                name = "req",
                methods = {HttpMethod.GET},
                authLevel = AuthorizationLevel.ANONYMOUS,
                route = "identity")
            HttpRequestMessage<Optional<String>> request,
            final ExecutionContext context) {

        String identityEndpoint = System.getenv("IDENTITY_ENDPOINT");
        String identityHeader = System.getenv("IDENTITY_HEADER");

        boolean hasManagedIdentity = identityEndpoint != null && identityHeader != null;

        context.getLogger().info("Identity probe: managed identity "
            + (hasManagedIdentity ? "available" : "not available"));

        String body = String.format(
            "{\"managedIdentity\":%s,\"identityEndpoint\":\"%s\"}",
            hasManagedIdentity,
            hasManagedIdentity ? "configured" : "not configured");

        return request.createResponseBuilder(HttpStatus.OK)
                .header("Content-Type", "application/json")
                .body(body)
                .build();
    }
}

File: apps/dotnet/Functions/IdentityProbeFunction.cs
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.Functions.Worker;
using Microsoft.Extensions.Logging;

namespace AzureFunctionsGuide.Functions;

/// <summary>
/// Managed identity status probe.
/// Route: GET /api/identity
/// </summary>
public class IdentityProbeFunction
{
    private readonly ILogger<IdentityProbeFunction> _logger;

    public IdentityProbeFunction(ILogger<IdentityProbeFunction> logger)
    {
        _logger = logger;
    }

    [Function("identityProbe")]
    public IActionResult Run(
        [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "identity")] HttpRequest req)
    {
        _logger.LogInformation("Identity probe requested");

        var identityEndpoint = System.Environment.GetEnvironmentVariable("IDENTITY_ENDPOINT");
        var hasManagedIdentity = !string.IsNullOrEmpty(identityEndpoint);

        return new OkObjectResult(new
        {
            managedIdentity = hasManagedIdentity,
            identityEndpoint = hasManagedIdentity ? "configured" : "not configured"
        });
    }
}
